feat: setup PostgreSQL with Docker configuration

Guzbyte-tech opened the pull request:
# 📝 Pull Request Title
Add Docker Setup for PostgreSQL Database

## 🛠️ Issue
Closes #98 

## 📖 Description
This pull request sets up a PostgreSQL database using Docker and Docker Compose for development and testing purposes. It includes data persistence and detailed setup instructions in the `apps/backend/postgre-setup.md` file.

## ✅ Changes made
All changes are in the `apps/backend` directory
- Added a docker-compose.yml file to configure and run a PostgreSQL container.
- Added a .envl file to store Postgre credentials.
- Configured environment variables for the database (POSTGRES_DB, POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_PORT).
Updated postgre-setup.md with step-by-step instructions for setting up and running the database.

## 🖼️ Media (screenshots/videos)
![image](https://github.com/user-attachments/assets/106cc225-769c-406f-a4d3-f8302ccbdf1a)




## 📜 Additional Notes
- Ensure Docker and Docker Compose are installed before using the setup.
- If port 5432 is in use, update the docker-compose.yml file to expose a different port (e.g., 5433).

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+
+services:
+  postgres:
+    image: postgres:latest
```
Change the image to `postgres:15-alpine`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+services:
+  postgres:
+    image: postgres:latest
+    container_name: project_db
```
Rename `project_db` to `safeswap_db`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+    image: postgres:latest
+    container_name: project_db
+    environment:
+      POSTGRES_DB: ${POSTGRES_DB:-project_database}
```
Rename `project_database` to `safeswap_database`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+    container_name: project_db
+    environment:
+      POSTGRES_DB: ${POSTGRES_DB:-project_database}
+      POSTGRES_USER: ${POSTGRES_USER:-project_user}
```
Rename `project_user` to `safeswap_user`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+    environment:
+      POSTGRES_DB: ${POSTGRES_DB:-project_database}
+      POSTGRES_USER: ${POSTGRES_USER:-project_user}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-project_password}
```
Rename `project_password` to `safeswap_password`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+      - postgres_data:/var/lib/postgresql/data
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER:-project_user} -d ${POSTGRES_DB:-project_database}"]
```
Here too

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+    ports:
+      - "${POSTGRES_PORT:-5432}:5432"
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
```
Rename `postgres_data` to `safeswap_data`

derianrddev commented on apps/backend/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+      retries: 5
+
+volumes:
+  postgres_data:
```
Here too

derianrddev commented on apps/backend/docker-compose.yml:
Add a blank line at the end of this file, please

derianrddev commented on apps/backend/.env.example:
Add a blank line at the end of this file, please

derianrddev reviewed (changes requested):
@Guzbyte-tech, Nice job! 🚀 Just a few minor tweaks to improve the setup:

1. Update the PostgreSQL image to `postgres:15-alpine` for a stable and lightweight version.
2. Replace all occurrences of "project" with "safewap" for consistency with the project name.
3. Add a blank line at the end of `.env.example` and `docker-compose.yml` to address the GitHub warning and follow best practices.

Great work overall! 👍

Guzbyte-tech commented:
- Updated the PostgreSQL image to postgres:15-alpine.
- Replace all occurrences of "project" with "safewap".
- I added a blank line at the end of the .env.example and docker-compose.yml to address the GitHub warning and follow best practices.

derianrddev reviewed (approved):
@Guzbyte-tech Thank you for making the changes. Everything looks good to me. Great job! 🚀

danielcdz reviewed (approved):
Awesome job
